CMakeLists.txt

```diff
@@ -18,6 +18,16 @@ add_custom_target(add-copyright
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   COMMENT "Add copyright")
 
+if(WIN32)
```

Comment:
I think this should be conditional on MSVC since it's Microsoft compiler flags that are modified.
Someone might want to compile using MinGW on Windows, and hence see GCC flags.

E.g. see the    if (MSVC) conditional in cmake/utils.cmak